tools/clang/tools/dxil2spv/lib/dxil2spv.cpp

```diff
@@ -37,43 +38,47 @@ Translator::Translator(CompilerInstance &instance)
       featureManager(diagnosticsEngine, spirvOptions),
       spvBuilder(spvContext, spirvOptions, featureManager) {}
 
-int Translator::Run(CComPtr<IDxcBlobEncoding> blob) {
-  const char *blobContext =
-      reinterpret_cast<const char *>(blob->GetBufferPointer());
-  unsigned blobSize = blob->GetBufferSize();
-
-  llvm::LLVMContext context;
-  llvm::SMDiagnostic err;
-  std::unique_ptr<llvm::MemoryBuffer> memoryBuffer;
-  std::unique_ptr<llvm::Module> module;
+int Translator::Run() {
+  // Read input file to memory buffer.
+  std::string filename = ci.getCodeGenOpts().MainFileName;
+  auto errorOrInputFile = llvm::MemoryBuffer::getFileOrSTDIN(filename);
+  if (!errorOrInputFile) {
+    emitError("Error reading %0: %1")
+        << filename << errorOrInputFile.getError().message();
+    return DXC_E_GENERAL_INTERNAL_ERROR;
+  }
+  llvm::MemoryBuffer *memoryBuffer = errorOrInputFile.get().release();
```

Comment:
I am not versed on the output of `llvm::MemoryBuffer::getFileOrSTDIN(..)` but I am curious about the side effect of referencing the MemoryBuffer after calling `release()`. Is it ok in terms of the use-after-free?

Reply:
There's no use-after-free risk, but moving the `std::unique_ptr` is preferable to releasing the raw pointer so I've switched it.